include/model.php

```diff
@@ -842,7 +842,15 @@ protected function get_languages_from_taxonomies() {
 		}
 
 		// Restore the right order after the first `array_reverse()`.
-		$terms_by_slug = array_reverse( $terms_by_slug );
+		// Also keep the terms order with the 'term_language' taxonomy first.
+		$terms_by_slug = array_reverse(
+			array_map(
+				function( $term_by_slug ) {
+					return array_reverse( $term_by_slug );
+				},
+				$terms_by_slug
+			)
+		);
```

Comment:
I think it can be simplified:
```suggestion
		$terms_by_slug = array_reverse(
			array_map( 'array_reverse', $terms_by_slug )
		);
```